src/System.Private.CoreLib/src/System/Diagnostics/Eventing/EventPipeController.cs

```diff
@@ -300,33 +305,87 @@ private static void SetProviderConfiguration(string strConfig, EventPipeConfigur
+            // String must be of the form "providerName:keywords:level:parameter,providerName:keywords:level:parameter..."
+            // where parameter is of the form: "key=value;key=value..."
+            string[] providers = strConfig.Split(
+                ProviderConfigDelimiter,
+                StringSplitOptions.RemoveEmptyEntries); // Remove "empty" providers.
             foreach (string provider in providers)
             {
-                string[] components = provider.Split(ConfigComponentDelimiter);
-                if (components.Length == 3)
-                {
-                    string providerName = components[0];
+                const char FilterDataDelimiter = ';';
+                string[] components = provider.Split(
+                    ConfigComponentDelimiter,
+                    StringSplitOptions.None); // Keep empty tokens
 
-                    // We use a try/catch block here because ulong.TryParse won't accept 0x at the beginning
-                    // of a hex string.  Thus, we either need to conditionally strip it or handle the exception.
-                    // Given that this is not a perf-critical path, catching the exception is the simpler code.
-                    ulong keywords = 0;
-                    try
-                    {
-                        keywords = Convert.ToUInt64(components[1], 16);
-                    }
-                    catch { }
+                string providerName = null;
+                ulong keywords = 0;
+                uint level = 0;
+                string filterData = null;
 
-                    uint level;
-                    if (!uint.TryParse(components[2], out level))
+                for (int i = 0; i < components.Length; ++i)
+                {
+                    switch(i)
                     {
-                        level = 0;
+                        case 0: // provider
+                            providerName = components[i];
+                            break;
+                        case 1: // keywords
+                            // We use a try/catch block here because ulong.TryParse won't accept 0x at the beginning
+                            // of a hex string.  Thus, we either need to conditionally strip it or handle the exception.
+                            // Given that this is not a perf-critical path, catching the exception is the simpler code.
+                            try
+                            {
+                                keywords = Convert.ToUInt64(components[i], 16);
+                            }
+                            catch { }
+                            break;
+                        case 2: // level
+                            uint.TryParse(components[i], out level);
+                            break;
+                        case 3: // parameter
+                            var dict = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase);
```

Comment:
You parse the string, just to put it back in the same format.   Yes, that does allow errors to be found quicker, but frankly not all errors, and realistically, tools will be writing this config file so they are in a better place to do checking.   I would remove all this code and simply set the filterData = components[i].  

As a side note, I would not use a loop but simply have a 
```
if (length > 0) { provider = ... } 
if (length > 1) { keywords = ... }
if (length > 2) { parameter= ...}
```
This mirrors the logic of what is going on more simply,  but I suppose that is more a style thing...